nighres/registration/embedded_syn.py

```diff
@@ -0,0 +1,255 @@
+import cbstools
+from ..io import load_volume, save_volume
+from ..utils import _output_dir_4saving, _fname_4saving, \
+                    _check_topology_lut_dir
+
+# convenience labels
+X=0
+Y=1
+Z=2
+T=3
+
+def embedded_syn(source_image, target_image, coarse_iterations=40, 
+                    medium_iterations=50, fine_iterations=40,
+					run_affine_first=False, cost_function='MutualInformation', 
+					interpolation='NearestNeighbor',
+                    save_data=False, output_dir=None,
+                    file_name=None):
+    """ Embedded SyN
+
+    Runs the Symmetric Normalization (SyN) algorithm of ANTs and formats the 
+    output deformations into voxel coordinate mappings as used in CBSTools 
+    registration and transformation routines.
+
+    Parameters
+    ----------
+    source_image: niimg
+        Image to register
+    target_image: niimg
+        Reference image to match
+    coarse_iterations: float
+        Number of iterations at the coarse level (default is 40)
+    medium_iterations: float
+        Number of iterations at the medium level (default is 50)
+    fine_iterations: float
+        Number of iterations at the fine level (default is 40)
+    run_affine_first: bool
+        Runs a step of affine registration before the non-linear step (default is False)
+    cost_function: {'CrossCorrelation', 'MutualInformation'}
+        Cost function for the registration (default is 'MutualInformation')
+    interpolation: {'NearestNeighbor', 'Linear'}
+        Interpolation for the registration (default is 'NearestNeighbor')
+    save_data: bool
+        Save output data to file (default is False)
+    output_dir: str, optional
+        Path to desired output directory, will be created if it doesn't exist
+    file_name: str, optional
+        Desired base name for output files with file extension
+        (suffixes will be added)
+
+    Returns
+    ----------
+    dict
+        Dictionary collecting outputs under the following keys
+        (suffix of output files in brackets)
+
+        * deformed_source (niimg): Deformed source image (_syn_def)
+        * mapping (niimg): Coordinate mapping from source to target (_syn_map)
+        * inverse (niimg): Inverse coordinate mapping from target to source (_syn_invmap) 
+
+    Notes
+    ----------
+    Port of the CBSTools Java module by Pierre-Louis Bazin. The main algorithm
+    is part of the ANTs software by Brian Avants and colleagues [1]_. The interfacing
+    with ANTs is performed through Nipype [2]_.
+
+    References
+    ----------
+    .. [1] Avants et al (2008), Symmetric diffeomorphic 
+       image registration with cross-correlation: evaluating automated labeling 
+       of elderly and neurodegenerative brain, Med Image Anal. 12(1):26-41
+    .. [2] Gorgolewski et al (2011) Nipype: a flexible, lightweight and extensible 
+       neuroimaging data processing framework in python. Front Neuroinform 5. 
+       doi:10.3389/fninf.2011.00013
+    """
+
```

Comment:
Include the import here: 

    try:
        from nipype.interfaces.ants import ANTS
        from nipype.interfaces.ants import ApplyTransforms
    except ImportError:
        print('Please install Nipype to use this function')
        return